mixer/pkg/server/args.go

```diff
@@ -0,0 +1,161 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"bytes"
+	"fmt"
+
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/il/evaluator"
+	mixerRuntime "istio.io/istio/mixer/pkg/runtime"
+	"istio.io/istio/mixer/pkg/template"
+)
+
+type Args struct {
+	// The templates to register.
+	Templates map[string]template.Info
+
+	// The adapters to use
+	Adapters []adapter.InfoFn
+
+	// The legacy adapters to use
+	LegacyAdapters []adapter.RegisterFn
+
+	// Maximum size of individual gRPC messages
+	MaxMessageSize uint
+
+	// Maximum number of outstanding RPCs per connection
+	MaxConcurrentStreams uint
+
+	// Maximum number of goroutines in the API worker pool
+	ApiWorkerPoolSize int
+
+	// Maximum number of goroutines in the adapter worker pool
+	AdapterWorkerPoolSize int
+
+	// Maximum number of entries in the expression cache
+	ExpressionEvalCacheSize int
+
+	// Port to use for Mixer's gRPC API
+	APIPort uint16
+
+	// Port to use for Mixer's Configuration API
+	ConfigAPIPort uint16
+
+	// Port to use for exposing mixer self-monitoring information
+	MonitoringPort uint16
+
+	// If true, each request to Mixer will be executed in a single go routine (useful for debugging)
+	SingleThreaded bool
+
+	// URL of zipkin collector (example: 'http://zipkin:9411/api/v1/spans'). This enables tracing for Mixer itself.
+	ZipkinURL string
+
+	// URL of jaeger HTTP collector (example: 'http://jaeger:14268/api/traces?format=jaeger.thrift'). This enables tracing for Mixer itself.
+	JaegerURL string
+
+	// Whether or not to log Mixer trace spans. This enables tracing for Mixer itself.
+	LogTraceSpans bool
+
+	// URL of the config store. May be fs:// for file system, or redis:// for redis url
+	ConfigStoreURL string
+
+	// URL of the config store. Use k8s://path_to_kubeconfig or fs:// for file system. If path_to_kubeconfig is empty, in-cluster kubeconfig is used.")
+	ConfigStore2URL string
+
+	// Kubernetes namespace used to store mesh-wide configuration.")
+	ConfigDefaultNamespace string
+
+	// Configuration fetch interval in seconds
+	ConfigFetchIntervalSec uint
+
+	// Attribute that is used to identify applicable scopes.
+	ConfigIdentityAttribute string
+
+	// The domain to which all values of the ConfigIdentityAttribute belong.
+	// For kubernetes services it is svc.cluster.local
+	ConfigIdentityAttributeDomain string
+
+	// Upper limit for String table size to purge at.
+	StringTablePurgeLimit int
+
+	// Deprecated
+	ServiceConfigFile string
+
+	// Deprecated
+	GlobalConfigFile string
+
+	// Supplies a string to use for service configuration, overrides ConfigStoreURL
+	ServiceConfig string
+
+	// Supplies a string to use for global configuration, overrides ConfigStoreURL
+	GlobalConfig string
+}
```

Comment:
aside: we should add a config option (at some point -- not this PR) to disable/enable self-monitoring bits (maybe you want to turn off rpc duration monitoring?).